test/e2e/apimachinery/chunking.go

```diff
@@ -101,4 +108,73 @@ var _ = SIGDescribe("Servers with support for API chunking", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(list.Items).To(HaveLen(numberOfTotalResources))
 	})
+
+	It("should support continue listing from the last key if the original version has been compacted away, though the list is inconsistent", func() {
+		ns := f.Namespace.Name
+		c := f.ClientSet
+		client := c.CoreV1().PodTemplates(ns)
+		By("retrieving the first page")
+		oneTenth := int64(numberOfTotalResources / 10)
+		opts := metav1.ListOptions{}
+		opts.Limit = oneTenth
+		list, err := client.List(opts)
+		Expect(err).ToNot(HaveOccurred())
+		firstToken := list.Continue
+		firstRV := list.ResourceVersion
+		framework.Logf("Retrieved %d/%d results with rv %s and continue %s", len(list.Items), opts.Limit, list.ResourceVersion, firstToken)
+		By("retrieving the second page until the token expires")
+		opts.Continue = firstToken
+		var inconsistentToken string
+		wait.Poll(20*time.Second, 2*storagebackend.DefaultCompactInterval, func() (bool, error) {
```

Comment:
This is a really questionable test.  I don't think this should be an e2e test, I think you need to be testing it as an integration test or in unit level.  Waiting 15 minutes in an e2e test is dumb.

Reply:
Hrm - that's about 3-4x longer than almost every other test except the very slow stateful sets.

Reply:
I guess I can buy it.